src/uct/ib/rc/verbs/rc_verbs_iface.c

```diff
@@ -122,6 +120,8 @@ static void uct_rc_verbs_handle_failure(uct_ib_iface_t *ib_iface, void *arg,
             "send completion with error: %s [qpn 0x%x wrid 0x%lx"
             "vendor_err 0x%x]\n%s", ibv_wc_status_str(wc->status), wc->qp_num,
             wc->wr_id, wc->vendor_err, peer_info);
+    ucs_arbiter_dispatch(&iface->tx.arbiter, 1, uct_rc_ep_process_pending,
+                         NULL);
```

Comment:
why needed? It is also called by `uct_rc_verbs_update_tx_res` above

Reply:
we split `uct_rc_verbs_update_tx_res` to not have `ucs_arbiter_dispatch`
we do dispatching of Arbiter after calling user's error handler

Reply:
can it introduce any ordering issues?

Reply:
yes, it could if user will call progress from error callback. but it is prohibited if I'm not mistaken

Reply:
actually you are right: we will fail on post_send assert
so - can you pls wrap "ucs_arbiter_dispatch(...)" call in a helper inline function? it repeats in multiple places

Reply:
progress is not really needed to create reordering - it is enough to send something. Bu this is also supposed to be prohibited, right?

Reply:
> Bu this is also supposed to be prohibited, right?

yep